client/client.go

```diff
@@ -0,0 +1,63 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/fabric8-services/fabric8-auth-client/auth"
+	goaclient "github.com/goadesign/goa/client"
+	"github.com/goadesign/goa/middleware/security/jwt"
+	errs "github.com/pkg/errors"
+)
+
+func NewAuthClient(hostURL string) (*AuthClient, error) {
+	u, err := url.Parse(hostURL)
+	if err != nil {
+		return nil, err
+	}
+
+	client := http.Client{}
```

Comment:
is cache needed? like this [notification_code](https://github.com/fabric8-services/fabric8-notification/blob/master/auth/client.go#L19)